Newtonsoft.Json Reference needs to match Cake.Core

#  Expected Behavior

The expected output should not contain warnings.

```
========================================
Default
========================================
Hello World

Task                          Duration
--------------------------------------------------
Default                       00:00:00.0074332
--------------------------------------------------
Total:                        00:00:00.0074332

```


## Current Behavior

```
(2701,12): warning CS1701: Assuming assembly reference 'Newtonsoft.Json, Version=11.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' used by 'Cake.Json' matches identity 'Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' of 'Newtonsoft.Json', you may need to supply runtime policy
(2707,12): warning CS1701: Assuming assembly reference 'Newtonsoft.Json, Version=11.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' used by 'Cake.Json' matches identity 'Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' of 'Newtonsoft.Json', you may need to supply runtime policy

========================================
Default
========================================
Hello World

Task                          Duration
--------------------------------------------------
Default                       00:00:00.0074332
--------------------------------------------------
Total:                        00:00:00.0074332
```

## Possible Solution

Update the Newtonsoft.Json reference to match Cake.Core

## Related Issues

https://github.com/cake-build/cake/issues/2582
https://github.com/cake-build/cake/issues/2606

## Steps to Reproduce (for bugs)

Using the following script json.cake script

```
#addin nuget:?package=Cake.Json&version=4.0.0
#addin nuget:?package=Newtonsoft.Json&version=11.0.1
// or #addin nuget:?package=Newtonsoft.Json&version=12.0.2

var target = Argument("target", "Default");
Task("Default")
  .Does(() => {
    Information("Hello World");
  });

RunTarget(target);

```
With .NET core 3.0 Installed Run.

```
dotnet tool install Cake.Tool --version 0.35.0
dotnet cake json.cake --bootstrap
dotnet cake json.cake
```


## Your Environment

dotnet --info                                                                

```
NET Core SDK (reflecting any global.json):                                                           
 Version:   3.0.100                                                                                   
 Commit:    04339c3a26                                                                                
                                                                                                      
Runtime Environment:                                                                                  
 OS Name:     Windows                                                                                 
 OS Version:  10.0.17763                                                                              
 OS Platform: Windows                                                                                 
 RID:         win10-x64                                                                               
 Base Path:   C:\Program Files\dotnet\sdk\3.0.100\                                                    
                                                                                                      
Host (useful for support):                                                                            
  Version: 3.0.0                                                                                      
  Commit:  7d57652f33                                                                                 
                                                                                                      
.NET Core SDKs installed:                                                                             
  1.0.0-preview2-003133 [C:\Program Files\dotnet\sdk]                                                 
  2.1.104 [C:\Program Files\dotnet\sdk]                                                               
  2.1.105 [C:\Program Files\dotnet\sdk]                                                               
  2.1.200 [C:\Program Files\dotnet\sdk]                                                               
  2.1.201 [C:\Program Files\dotnet\sdk]                                                               
  2.1.202 [C:\Program Files\dotnet\sdk]                                                               
  2.1.500 [C:\Program Files\dotnet\sdk]                                                               
  2.1.504 [C:\Program Files\dotnet\sdk]                                                               
  2.1.507 [C:\Program Files\dotnet\sdk]                                                               
  2.2.105 [C:\Program Files\dotnet\sdk]                                                               
  2.2.401 [C:\Program Files\dotnet\sdk]                                                               
  3.0.100 [C:\Program Files\dotnet\sdk]                                                               
                                                                                                      
.NET Core runtimes installed:                                                                         
  Microsoft.AspNetCore.All 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]            
  Microsoft.AspNetCore.All 2.1.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]            
  Microsoft.AspNetCore.All 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]           
  Microsoft.AspNetCore.All 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]           
  Microsoft.AspNetCore.All 2.2.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]            
  Microsoft.AspNetCore.All 2.2.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]            
  Microsoft.AspNetCore.All 2.2.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]            
  Microsoft.AspNetCore.App 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]            
  Microsoft.AspNetCore.App 2.1.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]            
  Microsoft.AspNetCore.App 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]           
  Microsoft.AspNetCore.App 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]           
  Microsoft.AspNetCore.App 2.2.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]            
  Microsoft.AspNetCore.App 2.2.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]            
  Microsoft.AspNetCore.App 2.2.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]            
  Microsoft.AspNetCore.App 3.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]            
  Microsoft.NETCore.App 2.0.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]                  
  Microsoft.NETCore.App 2.0.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]                  
  Microsoft.NETCore.App 2.0.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]                  
  Microsoft.NETCore.App 2.1.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]                  
  Microsoft.NETCore.App 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]                  
  Microsoft.NETCore.App 2.1.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]                  
  Microsoft.NETCore.App 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]                  
  Microsoft.NETCore.App 2.1.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]                  
  Microsoft.NETCore.App 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]                 
  Microsoft.NETCore.App 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]                 
  Microsoft.NETCore.App 2.2.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]                  
  Microsoft.NETCore.App 2.2.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]                  
  Microsoft.NETCore.App 2.2.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]                  
  Microsoft.NETCore.App 2.2.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]                  
  Microsoft.NETCore.App 2.2.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]                  
  Microsoft.NETCore.App 3.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]                  
  Microsoft.WindowsDesktop.App 3.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]    

```

--- a/README.md
+++ b/README.md
@@ -7,10 +7,10 @@ You can easily reference Cake.Json directly in your build script via a cake addi
 
 ```csharp
 #addin nuget:?package=Cake.Json
-#addin nuget:?package=Newtonsoft.Json&version=11.0.2
+#addin nuget:?package=Newtonsoft.Json&version=12.0.2
 ```
 
-NOTE: It's very important at this point in time to specify the Newtonsoft.Json package *and* the version _11.0.2_ for it.
+NOTE: It's very important at this point in time to specify the Newtonsoft.Json package *and* the version _12.0.2_ for it.
 
 ## Aliases
 

--- a/src/Cake.Json.Tests/Cake.Json.Tests.csproj
+++ b/src/Cake.Json.Tests/Cake.Json.Tests.csproj
@@ -5,9 +5,9 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Cake.Core" Version="0.33.0" />
-    <PackageReference Include="Cake.Testing" Version="0.33.0" />
-    <PackageReference Include="Newtonsoft.Json" Version="11.0.2" />
+    <PackageReference Include="Cake.Core" Version="0.35.0" />
+    <PackageReference Include="Cake.Testing" Version="0.35.0" />
+    <PackageReference Include="Newtonsoft.Json" Version="12.0.2" />
     <PackageReference Include="NSubstitute" Version="4.0.0" />
     <PackageReference Include="xunit" Version="2.4.1" />
     <PackageReference Include="xunit.runner.visualstudio" Version="2.4.1" />

--- a/src/Cake.Json/Cake.Json.csproj
+++ b/src/Cake.Json/Cake.Json.csproj
@@ -21,9 +21,9 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Cake.Core" Version="0.33.0" PrivateAssets="All" />
-    <PackageReference Include="Cake.Common" Version="0.33.0" PrivateAssets="All" />
-    <PackageReference Include="Newtonsoft.Json" Version="11.0.2" />
+    <PackageReference Include="Cake.Core" Version="0.35.0" PrivateAssets="All" />
+    <PackageReference Include="Cake.Common" Version="0.35.0" PrivateAssets="All" />
+    <PackageReference Include="Newtonsoft.Json" Version="12.0.2" />
   </ItemGroup>
 
 </Project>

